### PR TITLE
[WIP] Transaction Recovery Storage fixes from 3.1.0 and 3.1.1

### PR DESCRIPTION
**Pull request by DavidBoike**
CoreV6 implementation related to #164
Connects to #70 

Requires config of transaction recovery storage path and then sets up all the DTC settings correctly.

Connects to #136

### CODE REVIEW

**Comment by DavidBoike** [comment]
Ready for review @Particular/persistence-maintainers.

Of particular note, the changes I made to When_raven_session_is_provided since the ATT infrastructure should be allowed to create the DB, and NServiceBus should be allowed to initialize the DocStore, which means the Session can't be created right at the beginning of the test.

Also removed the public option to create a new DocStore in the ATT configuration class, to make it much harder for future tests to assume that's something that should be done.

File: src/NServiceBus.RavenDB/Internal/RavenTransactionRecoveryStorageCreator.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            throw new InvalidOperationException($"Unable to store RavenDB transaction recovery storage information. Specify a writeable directory using the `{RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey}` appSetting or .SetTransactionRecoveryStorageBasePath(basePath) option. `%LOCALAPPDATA%`, `%APPDATA%`, or `%PROGRAMDATA%` are good candidates. The same path can be shared by multiple endpoints.");
+        }
+
+        private IEnumerable<LocalDirectoryTransactionRecoveryStorage> TryToCreateFromConfiguredLocations(ReadOnlySettings settings)
+        {
+            var settingsBasePath = settings.GetOrDefault<string>(RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey);
+            yield return TryCreate(settingsBasePath,
+                "Unable to access RavenDB transaction recovery storage specified by `.SetTransactionRecoveryStorageBasePath(string basePath)`.");
+
+            var configBasePath = ConfigurationManager.AppSettings[RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey];
```
Comment (andreasohlund) [inline src/NServiceBus.RavenDB/Internal/RavenTransactionRecoveryStorageCreator.cs]: :-1: to provide an appsettings api (changing this in production == data loss)

File: src/NServiceBus.RavenDB/Internal/RavenTransactionRecoveryStorageCreator.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        private LocalDirectoryTransactionRecoveryStorage TryCreate(string basePath, string errorMessage)
+        {
+            if (basePath == null)
+            {
+                return null;
+            }
+
+            var fullPath = Path.Combine(Environment.ExpandEnvironmentVariables(basePath), this.suffixPath);
```
Comment (andreasohlund) [inline src/NServiceBus.RavenDB/Internal/RavenTransactionRecoveryStorageCreator.cs]: :-1: for this. Why not just ask the user to pass the path as is? (if we get this wrong users will loose data)

**Comment by DavidBoike** [comment]
Closing. Something different will happen.

**Pull request closed without merging**